src/fields2cover/route_planning/route_planner_base.cpp

```diff
@@ -107,10 +107,12 @@ F2CGraph2D RoutePlannerBase::createCoverageGraph(
         for (const auto& s2 : swaths2) {
           auto s2_s = s2.startPoint();
           auto s2_e = s2.endPoint();
-          g.addEdge(s1_s, s2_s, shortest_graph);
+          if (redirect_swaths) {
+            g.addEdge(s1_s, s2_s, shortest_graph);
+            g.addEdge(s1_e, s2_e, shortest_graph);
+          }
           g.addEdge(s1_s, s2_e, shortest_graph);
```

Comment:
I think connections are not properly made in this case. Be aware that `addEdge(a, b)` is bidirectional, so you are connection `a->b` and `b->a`. When reordering is not allowed, you only need `addDirectedEdge(s1_e, s2_s, shortest_graph)`

Reply:
Oh thanks, I did not know that. I made some changes to address this comment. Is there anything else that you know of that can simplify the search space to decrease runtime given that the user knows the direction/pattern of the swaths, but they are ok with having the order rearranged? For example, if they want the arrows  of the swaths to stay fixed in the same direction, but they just want the fastest/shortest route that goes through all the swaths.